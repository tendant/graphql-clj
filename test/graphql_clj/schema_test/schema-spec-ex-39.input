"""
A simple GraphQL schema which is well described.
"""
type Query {
  """
  Translates a string from a given language into a different language.
  """
  translate(
    "The original language that `text` is provided in."
    fromLanguage: Language

    "The translated language to be returned."
    toLanguage: Language

    "The text to be translated."
    text: String
  ): String
}

"""
The set of languages supported by `translate`.
"""
enum Language {
  "English"
  EN

  "French"
  FR

  "Chinese"
  CH
}