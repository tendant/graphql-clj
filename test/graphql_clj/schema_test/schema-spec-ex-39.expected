{:objects
 {:Query
  {:description
   "\"\"\"\nA simple GraphQL schema which is well described.\n\"\"\"",
   :fields
   {:translate
    {:descritpion
     "\"\"\"\n  Translates a string from a given language into a different language.\n  \"\"\"",
     :args
     {:fromLanguage
      {:description
       "\"The original language that `text` is provided in.\"",
       :type :Language},
      :toLanguage
      {:description "\"The translated language to be returned.\"",
       :type :Language},
      :text
      {:description "\"The text to be translated.\"", :type String}},
     :type String}}}},
 :enums
 {:Language
  {:description
   "\"\"\"\nThe set of languages supported by `translate`.\n\"\"\"",
   :values
   [{:description "\"English\"", :enum-value :EN}
    {:description "\"French\"", :enum-value :FR}
    {:description "\"Chinese\"", :enum-value :CH}]}}}
